packages/serverless/src/domain/use-case/prover.ts

```diff
@@ -15,7 +9,24 @@ import { ClientSession } from 'mongodb';
 import { Field } from 'o1js';
 
 import { ModelMetadataDocument } from '@model';
-import { buildSchema } from './schema';
+import { Schema, TValidatedDocument } from './schema';
+
+// For prover param use-case
+export type TParamProve = {
+  databaseName: string;
+  collectionName: string;
+  docId: string;
+};
+
+export type TParamProveCreate = TParamProve & {
+  document: TValidatedDocument;
+};
+
+export type TParamProveUpdate = TParamProve & {
+  newDocument: TValidatedDocument;
```

Comment:
should be move to common

Reply:
these are serverless only, I think we should keep it there until future APIs need them. otherwise TValidatedDocument has to be in common too, which is implementation detail and leaky abstraction